pkg/summarizer/flakiness_test.go

```diff
@@ -0,0 +1,460 @@
+					{
+						Started: 0,
+					},
+					{
+						Started: 1,
+					},
+					{
+						Started: 2,
+					},
+					{
+						Started: 2,
+					},
+				},
+				Rows: []*state.Row{
+					{
+						Name: "test_1",
+						Results: []int32{
+							state.Row_Result_value["PASS"], 1,
+							state.Row_Result_value["FAIL"], 1,
+							state.Row_Result_value["FLAKY"], 1,
+							state.Row_Result_value["FAIL"], 1,
+						},
+						Messages: []string{
+							"",
+							"",
+							"",
+							"infra_fail_1",
+						},
+					},
+				},
+			},
+			startTime: 0,
+			endTime:   2,
+			expected: []Result{
+				{
+					name:             "test_1",
+					passed:           1,
+					failed:           1,
+					flakyCount:       1,
+					averageFlakiness: 50.0,
+					failedInfraCount: 1,
+					infraFailures: map[string]int{
+						"infra_fail_1": 1,
+					},
+				},
+			},
+		},
+		{
+			name: "grid with no analyzed results produces empty result list",
+			grid: &state.Grid{
+				Columns: []*state.Column{
+					{
+						Started: -1,
+					},
+					{
+						Started: 1,
+					},
+					{
+						Started: 2,
+					},
+					{
+						Started: 2,
+					},
+				},
+				Rows: []*state.Row{
+					{
+						Name: "test_1",
+						Results: []int32{
+							state.Row_Result_value["NO_RESULT"], 4,
+						},
+						Messages: []string{
+							"this message should not show up in results_0",
+							"this message should not show up in results_1",
+							"this message should not show up in results_2",
+							"this message should not show up in results_3",
+						},
+					},
+				},
+			},
+			startTime: 0,
+			endTime:   2,
+			expected:  []Result{},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if actual := parseGrid(tc.grid, tc.startTime, tc.endTime); !reflect.DeepEqual(actual, tc.expected) {
+				t.Errorf("\nactual %+v \n!= \nexpected %+v", actual, tc.expected)
+			}
+		})
+	}
+}
+
+func getTypicalResultArray() []Result {
+	return []Result{
+		{
+			name:             "//test1 - [env1]",
+			passed:           1,
+			failed:           1,
+			flakyCount:       1,
+			averageFlakiness: 50.0,
+			failedInfraCount: 1,
+			infraFailures: map[string]int{
+				"infra_fail_1": 1,
+			},
+		},
+	}
+}
+
+func TestNaiveFlakiness(t *testing.T) {
+	cases := []struct {
+		name      string
+		results   []Result
+		minRuns   int
+		startDate int
+		endDate   int
+		tab       string
+		expected  Healthiness
+	}{
+		{
+			name:      "typical case returns expected healthiness",
+			results:   getTypicalResultArray(),
+			minRuns:   -1,
+			startDate: 0,
+			endDate:   2,
+			tab:       "tab1",
+			expected: Healthiness{
+				startDate: 0,
+				endDate:   2,
+				tests: []TestInfo{
+					{
+						name:               "//test1",
+						env:                "env1",
+						totalRuns:          2,
+						totalRunsWithInfra: 3,
+						passedRuns:         1,
+						failedRuns:         1,
+						failedInfraRuns:    1,
+						flakyRuns:          1,
+						flakiness:          50,
```

Comment:
Does this need to be 50.0?

Reply:
I ended up adding comments + a TODO in the original function that sets the value that I would update the functionality later. For now it defaults to 50.0.